File: app/models/document.py
from datetime import datetime

from sqlalchemy import Column, String, Integer, ForeignKey, Date, DateTime
from sqlalchemy.orm import relationship

from app.database import Base


class Document(Base):
    __tablename__ = "documents"

    id = Column(Integer, primary_key=True, index=True)
    member_id = Column(Integer, ForeignKey("family_members.id"))
    document_type = Column(String, nullable=False)
    custom_document_name = Column(String, nullable=True)
    file_path = Column(String, nullable=False)
    issue_date = Column(Date, nullable=True)
    expiry_date = Column(Date, nullable=True)
    reminder_days_before = Column(Integer, nullable=False)
    uploaded_at = Column(DateTime, default=datetime.utcnow)

    member = relationship("FamilyMember", back_populates="documents")

File: frontend/src/types/document.ts
export interface DocumentItem {
  id: number;
  member_id: number;
  document_type: string;
  custom_document_name: string | null;
  file_path: string;
  issue_date: string | null;
  expiry_date: string | null;
  reminder_days_before: number;
  uploaded_at: string;
  extracted_data: Record<string, string> | null;
  status: string;
}